### DISCUSSION TITLE
Not loading, (CSP)

### DISCUSSION THREAD

**Discussion by IcyAlmond** [tsukasa/BdBrowser#18] [Problems and issues]
It's not loading for me, I think it's because of CSP but everyone else seems to have it working alright so i'm not sure what's going on.
<details>
<summary>Chromium</summary>

![image](https://user-images.githubusercontent.com/67372293/221392712-58f537e5-8dff-4408-904e-6d89f3b4cee2.png)
</details>
<details>
<summary>Firefox</summary>

![image](https://user-images.githubusercontent.com/67372293/221392722-fda8c07e-ff7a-4552-af64-67bedc9850e6.png)
</details>

Also, loading the addon on Firefox warns me with `Reading manifest: Invalid extension permission: declarativeNetRequestWithHostAccess`
I've also tried stuff like https://addons.mozilla.org/en-US/firefox/addon/cors-everywhere/ to no avail

**Comment by tsukasa** [comment]
Greetings,

please ensure the service worker is running. Your screenshot suggests that the declarativeNetRequest rules for stripping Content-Security-Policy headers within the worker have not been applied and that the shim for the `request` module does not run properly. There is little more to say with the available data, and I cannot reproduce the issue on Chrome, Edge or Wavebox.

Conceptually the service worker is intended to prevent this exact scenario. It registers declarativeNetRequest rules at the start that will strip CSP headers from responses to Discord's main page frame. Furthermore, Node shims that like `http`, `fetch` and `request` will route their requests through the service worker to ensure they do not get dropped.

Unfortunately, Firefox is not supported at this time due to a lacking implementation of Google's Manifest V3 standard. I experimented with workarounds but found that maintenance would quickly become an issue. I am hopeful that Mozilla's continued efforts to implement the missing specs will enable the use of the extension on Firefox in the future.
